src/controllers/commands_controller.js

```diff
@@ -114,28 +114,48 @@ CommandsController.post('/', function (request, response) {
           .then(function(thingBody) {
             let jsonBody = JSON.parse(thingBody);
             let match = payload.param.toUpperCase();
+            let thingfound = false; 
             for (var i = 0; i < jsonBody.length; i++) {
               var obj = jsonBody[i];
               let name = obj.name.toUpperCase();
               if (name == match) {
-                payload.href = obj.properties.on.href;
+                thingfound = true;
                 break;
               }
             }
-            if (payload.href != '') {
+            if (thingfound) {
               // Returning 201 to signify that the command was mapped to an
               // intent and matched a 'thing' in our list.  Return a response to
               // caller with this status before the command finishes execution
               // as the execution can take some time (e.g. blinds)
-              response.status(201).send(JSON.stringify(
-                {'message': 'Command Created'}));
-              iotOptions.uri = CommandsController.gatewayHref + payload.href;
+              if ((payload.param3 != null) && 
+                (payload.param3 != ''  && obj.properties.hue != '')) {
+                var colorname_to_hue = {red:0, orange:8500, yellow:17000, 
+                  green:25500, white:34000, blue:46920, purple:48000, 
+                  magenta:54000, pink:60000};
+                if (!colorname_to_hue[payload.param3]) {
+                  response.status(404).send(JSON.stringify(
+                    {'message': 'Hue color not found'}));
+                    return;
```

Comment:
nit: this should be indented one less level

Reply:
fixed